### PR TITLE
Item and Item Socket Props

### PR DESCRIPTION
**Pull request by Hawkbat**
## Major features
- Custom items and item sockets, as an extension of the existing details. Interactivity can be added without code by setting dialogue conditions with an item's `pickupCondition` or socket's `insertCondition`, and then turning other details on or off using their `activationCondition`.

## Minor features
- Added API methods for looking up translated strings for localization.

### CODE REVIEW

File: NewHorizons/Builder/Props/ItemBuilder.cs [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+            }
+
+            socket.ItemType = itemType;
+            socket.UseGiveTakePrompts = info.useGiveTakePrompts;
+            socket.InsertCondition = info.insertCondition;
+            socket.ClearInsertConditionOnRemoval = info.clearInsertConditionOnRemoval;
+            socket.RemovalCondition = info.removalCondition;
+            socket.ClearRemovalConditionOnInsert = info.clearRemovalConditionOnInsert;
+
+            Delay.FireInNUpdates(() =>
```
Comment (JohnCorby) [inline NewHorizons/Builder/Props/ItemBuilder.cs:150]: why the 2 frame delay? lol

File: NewHorizons/Builder/Props/ItemBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+                        {
+                            NHLogger.LogError($"Could not find a socket to parent item {itemName} to at path {socketGO}");
+                        }
+                    }
+                    else
+                    {
+                        NHLogger.LogError($"Could not find a socket to parent item {itemName} to at path {socketGO}");
+                    }
+                }
+            }, 1);
```
Comment (JohnCorby) [inline NewHorizons/Builder/Props/ItemBuilder.cs:114]: FireOnNextUpdate would be more clear here (heavy nitpick LOL)

File: NewHorizons/Builder/Props/DetailBuilder.cs [hunk 3]
```diff
@@ -52,24 +52,16 @@ public static GameObject GetSpawnedGameObjectByDetailInfo(DetailInfo detail)
         /// <summary>
         /// Create a detail using an asset bundle or a path in the scene hierarchy of the item to copy.
         /// </summary>
-        public static GameObject Make(GameObject go, Sector sector, IModBehaviour mod, DetailInfo detail)
```
Comment (JohnCorby) [inline NewHorizons/Builder/Props/DetailBuilder.cs:55]: why the rename? should the other Make's be renamed?

File: NewHorizons/Builder/Props/WarpPadBuilder.cs [hunk 4]
```diff
@@ -122,13 +123,13 @@ public static void Make(GameObject planetGO, Sector sector, NomaiWarpReceiverInf
 
             if (info.computer != null)
             {
-                CreateComputer(planetGO, sector, info.computer, receiver);
+                CreateComputer(planetGO, sector, mod, info.computer, receiver);
             }
         }
 
-        public static void Make(GameObject planetGO, Sector sector, NomaiWarpTransmitterInfo info)
```
Comment (JohnCorby) [inline NewHorizons/Builder/Props/WarpPadBuilder.cs:129]: is it worth commenting that mod is only used for Items? or is that obvious enough?

**Review by JohnCorby — APPROVED** [review]

File: NewHorizons/Builder/Props/ItemBuilder.cs (hunk 1 above)
Comment (Hawkbat) [reply NewHorizons/Builder/Props/ItemBuilder.cs:150]: To make sure it runs after the similarly delayed method in for items; I only want sockets to consider themselves empty if no items get inserted into them on scene startup.

File: NewHorizons/Builder/Props/ItemBuilder.cs (hunk 2 above)
Comment (Hawkbat) [reply NewHorizons/Builder/Props/ItemBuilder.cs:114]: Fair enough, I copied and pasted from the socket code lol

File: NewHorizons/Builder/Props/DetailBuilder.cs (hunk 3 above)
Comment (Hawkbat) [reply NewHorizons/Builder/Props/DetailBuilder.cs:55]: By adding the mod param it made it identical in signature to another overload, so I had to merge them one way or another. I decided to rename the params on the shorter one so there'd be fewer overall changes.

File: NewHorizons/Builder/Props/WarpPadBuilder.cs (hunk 4 above)
Comment (Hawkbat) [reply NewHorizons/Builder/Props/WarpPadBuilder.cs:129]: Currently it's only used for item sounds, but there's always the possibility that it'll be used to look up other kinds of resources (textures, XMLs, whatever), so I wouldn't want to put in comments that could become incorrect in the future.

**Review by MegaPiggy — APPROVED** [review]

**Pull request merged**